plasma_framework/migrations/5_deploy_and_register_payment_exit_game.js

```diff
@@ -0,0 +1,90 @@
+    const piggybackInFlightExit = await PaymentPiggybackInFlightExit.new();
+    const challengeInFlightExitNotCanonical = await PaymentChallengeIFENotCanonical.new();
+    const challengeIFEInputSpent = await PaymentChallengeIFEInputSpent.new();
+    const challengeIFEOutput = await PaymentChallengeIFEOutputSpent.new();
+    const processInFlightExit = await PaymentProcessInFlightExit.new();
+    await PaymentExitGame.link('PaymentStartStandardExit', startStandardExit.address);
+    await PaymentExitGame.link('PaymentChallengeStandardExit', challengeStandardExit.address);
+    await PaymentExitGame.link('PaymentProcessStandardExit', processStandardExit.address);
+    await PaymentExitGame.link('PaymentStartInFlightExit', startInFlightExit.address);
+    await PaymentExitGame.link('PaymentPiggybackInFlightExit', piggybackInFlightExit.address);
+    await PaymentExitGame.link('PaymentChallengeIFENotCanonical', challengeInFlightExitNotCanonical.address);
+    await PaymentExitGame.link('PaymentChallengeIFEInputSpent', challengeIFEInputSpent.address);
+    await PaymentExitGame.link('PaymentChallengeIFEOutputSpent', challengeIFEOutput.address);
+    await PaymentExitGame.link('PaymentProcessInFlightExit', processInFlightExit.address);
+
+    // deploy exit game
+    const outputGuardHandlerRegistry = await OutputGuardHandlerRegistry.new();
+    const spendingConditionRegistry = await SpendingConditionRegistry.new();
+    const stateVerifier = await PaymentTransactionStateTransitionVerifier.new();
+    const plasmaFramework = await PlasmaFramework.deployed();
+    const ethVaultAddress = await plasmaFramework.vaults(VAULT_ID.ETH);
+    const erc20VaultAddress = await plasmaFramework.vaults(VAULT_ID.ERC20);
+    const paymentExitGame = await PaymentExitGame.new(
+        plasmaFramework.address,
+        ethVaultAddress,
+        erc20VaultAddress,
+        outputGuardHandlerRegistry.address,
+        spendingConditionRegistry.address,
+        stateVerifier.address,
+        TX_TYPE.PAYMENT,
+    );
+
+    // handle output guard handler
+    const paymentOutputGuardHandler = await PaymentOutputGuardHandler.new(OUTPUT_TYPE.PAYMENT);
+    await outputGuardHandlerRegistry.registerOutputGuardHandler(
+        OUTPUT_TYPE.PAYMENT, paymentOutputGuardHandler.address,
+    );
+    await outputGuardHandlerRegistry.renounceOwnership();
+
+    // handle spending condition
+    const paymentToPaymentCondition = await PaymentOutputToPaymentTxCondition.new(
+        plasmaFramework.address, OUTPUT_TYPE.PAYMENT, TX_TYPE.PAYMENT,
+    );
+    const paymentToPaymentV2Condition = await PaymentOutputToPaymentTxCondition.new(
+        plasmaFramework.address, OUTPUT_TYPE.PAYMENT, TX_TYPE.PAYMENT_V2,
+    );
+    await spendingConditionRegistry.registerSpendingCondition(
+        OUTPUT_TYPE.PAYMENT, TX_TYPE.PAYMENT, paymentToPaymentCondition.address,
+    );
+    await spendingConditionRegistry.registerSpendingCondition(
```

Comment:
@thec00n one update on the dependency of spending condition registry that I missed previously. We are pre-defining how `PaymentTxV2` can spend `PaymentTx`. As a result, we can do soft upgrade by deploying `PaymentExitGameV2` in the future and ask users to transfer from `PaymentTx` to `PaymentTxV2`.